aio_pika/connection.py

```diff
@@ -1,114 +1,125 @@
 import asyncio
 import logging
 from functools import partial
-from typing import Optional, Type, TypeVar
+from types import TracebackType
+from typing import Any, Callable, Dict, Optional, Tuple, Type, TypeVar, Union
+from warnings import warn
 
 import aiormq
+from aiormq.abc import ExceptionType
 from aiormq.tools import censor_url
 from yarl import URL
 
+# For migration from 6.x to 7.x
+from .abc import ConnectionType  # noqa
```

Comment:
I didn't find any other way to tell the user that the import from this file is deprecated, so I removed it from `__all__` and now only the user linter (or IDE) will tell him about it. But I tried to ensure that the user code did not break during migration.

Reply:
As I said earlier, - 7.0 is a new major version so that is normal if it will break something.

Reply:
Yes but really sad when you upgrade a library and have to fix all your code and run all tests (etc.) to get it done. 

In my opinion, people love upgrades but really hate just technical changes which borns incompatible changes.

Moving the type definitions is not impacted to the memory consumption performance or whatever, and for me, it's a question of upgrade adoption. I have no plans for support 6.X branch right now cause I want to provide the simplest upgrade way.

This release contains changes in library core and promises minimal public API changes.

Probably you are right and I have to write these promises explicitly in the CHANGELOG.